### PR TITLE
fix(cli): allow workspaces in non-git projects and store relative paths in add

### PR DESCRIPTION
**Pull request by willosborne**
## Description

Fixes two of the five issues from #3031 ("Workspace command suite issues"):

- `calm workspace` commands (`init`, `list`, `show`, `switch`, `clean`, `push`, `bump`) required a `.git` directory and hard-failed otherwise. A new `findProjectRoot()` resolver falls back to the given directory (or cwd) when there's no git repo, instead of requiring a fake `.git`.
- `workspace add` stored the full absolute path to a referenced file in the manifest. It now stores a path relative to the bundle directory, matching the convention already used for `--copy`'d files.

The remaining three items in #3031 (standards support in `add`/`new`, stale `new` templates, and fully interactive `rm`) are not part of this PR — tracked separately.

## Type of Change
- [x] 🐛 Bug fix (non-breaking change which fixes an issue)

## Affected Components
- [x] CLI (`cli/`)

## Testing
- [x] I have tested my changes locally
- [x] I have added/updated unit tests
- [x] All existing tests pass

## Checklist
- [x] My commits follow the [conventional commit format](https://www.conventionalcommits.org/)
- [x] I have updated documentation if necessary
- [x] I have added tests for my changes (if applicable)
- [x] My changes follow the project's coding standards

### CODE REVIEW

**Review by rocketstack-matt — COMMENTED** [review]
A question around how you expect this edge case to work . . .

File: cli/src/workspace-resolver.ts [hunk 1]
```diff
@@ -27,10 +27,20 @@ export function findGitRoot(startPath?: string): string | null {
     return null;
 }
 
+/**
+ * Resolve the root directory workspace state should live under: the git root if one
+ * exists, otherwise startPath (or the cwd) itself. This lets workspaces be used from
+ * folders that aren't git repositories, without walking further up the filesystem
+ * when there's no git boundary to stop at.
+ */
+export function findProjectRoot(startPath?: string): string {
```
Comment (rocketstack-matt) [inline cli/src/workspace-resolver.ts:68]: Edge case: `findProjectRoot` re-resolves on every invocation, so a workspace created in a non-git directory becomes silently unreachable if a `.git` later appears in a parent directory — `list`/`show`/`switch`/`add` will start looking under the new (higher) root and report "No workspaces found" even though the workspace still exists on disk at the original path.

Repro: `workspace init` in `/a/b` (no git anywhere) → works; `git init` at `/a` → `workspace list` from `/a/b` now reports nothing found.

Not a blocker for this PR's scope, but worth a note or a defensive check (e.g. prefer a project root that already has a `.calm-workspace` over the git root) as a follow-up. (non-blocking nit)

Comment (willosborne) [reply cli/src/workspace-resolver.ts:68]: good shout. let me fix that

Comment (willosborne) [reply cli/src/workspace-resolver.ts:68]: Fixed in 960f61f0 — findProjectRoot now prefers the nearest existing `.calm-workspace` over the git root, bounded to the current git repo so it doesn't accidentally pick up an unrelated ancestor project's workspace. Added tests covering your repro plus the nested-subdirectory variant.

Comment (willosborne) [reply cli/src/workspace-resolver.ts:68]: fixed

**Review by rocketstack-matt — APPROVED** [review]
Verified: tests (137/137), lint, and build all pass on 960f61f0. The earlier edge case (workspace unreachable once a .git appears higher up) is fixed with good regression coverage.
